pkg/providers/v1/variant/variant.go

```diff
@@ -0,0 +1,88 @@
+package variant
+
+import (
+	"fmt"
+	"sync"
+
+	v1 "k8s.io/api/core/v1"
+	cloudprovider "k8s.io/cloud-provider"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+
+	"k8s.io/cloud-provider-aws/pkg/providers/v1/config"
+	"k8s.io/cloud-provider-aws/pkg/providers/v1/iface"
+)
+
+var variantsLock sync.Mutex
```

Comment:
I think we need a https://pkg.go.dev/sync#RWMutex since we'll have multiple routines calling `IsVariantNode` while writes will only happen during init-s

Reply:
only during `init`

Reply:
i believe all init's are called in a specific order